ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -904,3 +910,194 @@ def test_ipahealthcheck_trust_agent_member_check(self):
         for check in data:
             assert check["result"] == "SUCCESS"
             assert check["kw"]["key"] == self.master.hostname
+
+
+class TestIpaHealthCheckWithExternalCA(IntegrationTest):
+    """
+    Tests to run and check whether ipa-healthcheck tool
+    reports correct status when IPA server is configured
+    with external CA.
+    """
+
+    @classmethod
+    def install(cls, mh):
+        result = install_server_external_ca_step1(cls.master)
+        assert result.returncode == 0
+        root_ca_fname, ipa_ca_fname = tasks.sign_ca_and_transport(
+            cls.master, paths.ROOT_IPA_CSR, ROOT_CA, IPA_CA
+        )
+
+        install_server_external_ca_step2(
+            cls.master, ipa_ca_fname, root_ca_fname
+        )
+        tasks.kinit_admin(cls.master)
+
+    def test_ipahealthcheck_crlmanagercheck(self):
+        """
+        Test for IPACRLManagerCheck
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.roles", "IPACRLManagerCheck"
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] is True
+
+    @pytest.fixture()
+    def getcert_ca(self):
+        """
+        Fixture to remove and add ca using getcert command.
+        """
+        self.master.run_command(
+            ["getcert", "remove-ca", "-c", "dogtag-ipa-ca-renew-agent"]
+        )
+        yield
+        self.master.run_command(
+            [
+                "getcert",
+                "add-ca",
+                "-c",
+                "dogtag-ipa-ca-renew-agent",
+                "-e",
+                paths.DOGTAG_IPA_CA_RENEW_AGENT_SUBMIT,
+            ]
+        )
+        
+    def test_ipahealthcheck_certmongerca(self, getcert_ca):
+        """
+        Test for IPACertmongerCA
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPACertmongerCA",
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == "dogtag-ipa-ca-renew-agent":
+                assert check["result"] == "ERROR"
+                assert (
+                    check["kw"]["msg"]
+                    == "Certmonger CA 'dogtag-ipa-ca-renew-agent' missing"
+                )
+
+    @pytest.fixture()
+    def rename_httpd_cert(self):
+        """
+        Fixture to rename http cert and revert the change.
+        """
+        self.master.run_command(
+            ["mv", paths.HTTPD_CERT_FILE, paths.HTTPD_CERT_FILE + ".old"]
+        )
+        yield
+        self.master.run_command(
+            ["mv", paths.HTTPD_CERT_FILE + ".old", paths.HTTPD_CERT_FILE]
+        )
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPAOpenSSLChainValidation"
+        )
```

Comment:
This doesn't need to be in the fixture.

Reply:
removed the above mentioned code from fixture